File: 74-SearchA2dMatrix/74-SearchA2dMatrix.c
// Last updated: 6/6/2026, 10:26:59 PM
bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target)
{
        int row = matrixSize;
        int col = matrixColSize[0];

        int low = 0;
        int high = row*col -1;

        while(low<=high)
        {
            int mid = (low+high)/2;
            int r = mid / col;
            int c = mid % col;

            if(matrix[r][c] == target)
            {
                return true;
            }
            else if(matrix[r][c]<target)
            {
                low = mid+1;
            }
            else
            {
                high = mid-1;
            }

        }
        return false;

}

File: 74-SearchA2dMatrix/74-SearchA2dMatrix.py
# Last updated: 6/28/2026, 12:42:50 PM
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        # flag = True
        for i in matrix:
            low = 0 
            high = len(i) -1
            while(low <= high):
                mid = (low + high) // 2
                if(i[mid] == target):
                    return True
                elif(i[mid] < target):
                    low = mid + 1
                else:
                    high = mid - 1
        return False
            
            

        